filter/series_by_tag.go

```diff
@@ -63,7 +72,7 @@ func transformWildcardToRegexpInSeriesByTag(input string) (string, bool) {
 	}
 
 	if isTransformed {
-		result += "$"
+		result = fmt.Sprintf("^%s$", result)
```

Comment:
May be simple strinng concat is better (Sprintf with dynamic format string has bad perfomance).

Reply:
This is true, thanks. I've fixed in new commit.